### PR TITLE
[tcat] implementation of tcat disconnect command

### PR DESCRIPTION
**Pull request by canisLupus1313**
Implementation of tcat disconnect command in python client.

### CODE REVIEW

**Comment by canisLupus1313** [comment]
@arnulfrupp @EskoDijk FYI

File: tools/tcat_ble_client/cli/base_commands.py [hunk 1]
```diff
@@ -379,3 +384,14 @@ async def execute_default(self, args, context):
         else:
             print('Secure channel not established.')
             await ble_stream.disconnect()
+        return CommandResultNone()
+
+
+class DisconnectCommand(Command):
+
+    def get_help_string(self) -> str:
+        return 'Disconnect from device'
```
Comment (EskoDijk) [inline tools/tcat_ble_client/cli/base_commands.py:393]:
```suggestion
        return 'Disconnect client from TCAT device'
```

**Review by EskoDijk — CHANGES_REQUESTED** [review]
Looks good, a few string updates proposed and one variable name change.

**Review by abtink — APPROVED** [review]
LGTM

File: tools/tcat_ble_client/cli/base_commands.py [hunk 2]
```diff
@@ -352,7 +356,8 @@ def get_help_string(self) -> str:
         return 'Perform scan for TCAT devices.'
 
     async def execute_default(self, args, context):
-        if not (context['ble_sstream'] is None):
+        if 'ble_sstream' in context and context['ble_stream'] is not None:
```
[the remark this answers is not in the document]
Comment (canisLupus1313) [reply tools/tcat_ble_client/cli/base_commands.py]: It should. thanks for spotting.

File: tools/tcat_ble_client/cli/base_commands.py [hunk 3]
```diff
@@ -379,3 +384,14 @@ async def execute_default(self, args, context):
...
+        return CommandResultNone()
+
+
+class DisconnectCommand(Command):
+
+    def get_help_string(self) -> str:
+        return 'Disconnect client from TCAT device'
+
+    async def execute_default(self, args, context):
+        await context['ble_sstream'].close()
```
Comment (maciejbaczmanski) [inline tools/tcat_ble_client/cli/base_commands.py:396]: should we check if context exists here as well?

**Review by maciejbaczmanski — APPROVED** [review]
one suggestion

File: tools/tcat_ble_client/cli/base_commands.py (hunk 3 above)
Comment (canisLupus1313) [reply tools/tcat_ble_client/cli/base_commands.py:399]: great catch thanks.

**Review by EskoDijk — APPROVED** [review]
Thanks, LGTM - sorry for the delay in my response!

**Review by jwhui — APPROVED** [review]

**Pull request merged**